### PR TITLE
Simple rate limiting mechanism.

### PR DESCRIPTION
**Pull request by Marudny**
Simple nginx rate limiting mechanism.

Signed-off-by: Artur Marud <artur@status.im>

### CODE REVIEW

File: ansible/roles/ipfs-proxy/templates/gateway.conf.j2 [hunk 1]
```diff
@@ -20,6 +20,16 @@ server {
   # Set a size limit for uploads
   client_max_body_size {{ ipfs_proxy_api_exposed_max_body_size }};
 
+  {% if ipfs_proxy_rate_limit %}
```
Comment (jakubgs) [inline ansible/roles/ipfs-proxy/templates/gateway.conf.j2:23]: I'm not sure we really need the conditionals here. There is no instance of IPFS deployment where we would disable rate limiting, so the condition is kinda useless at this moment. I'd drop it.

Also, if I'm naming boolean flags like that I'd end them with `_enabled`, so it's clear it's a boolean.

**Review by jakubgs — CHANGES_REQUESTED** [review]
Since the IPFS instance is hosted behind CloudFlare proxy:
https://github.com/status-im/infra-ipfs/blob/bdf0964b5316e098760d94b4802a4385f725a377/ipfs.tf#L29-L36
This rate limiting will the done based on `$binary_remote_addr` which would be the IPs of CloudFlare proxy servers, and not actual clients. See how we do this in `dev-pages`:
https://github.com/status-im/infra-misc/blob/master/ansible/roles/dev-pages/templates/limits.conf.j2

**Comment by jakubgs** [comment]
Also, you might want to check and update the IPs list from CloudFlare: https://www.cloudflare.com/ips/

The list we have in `dev-pages` might be slightly out of date.

File: ansible/roles/ipfs-proxy/templates/gateway.conf.j2 (hunk 1 above)
Comment (Marudny) [reply ansible/roles/ipfs-proxy/templates/gateway.conf.j2]: Sure, I've removed the conditionals. Took note regarding naming convention.

File: ansible/roles/ipfs-proxy/defaults/main.yml [hunk 2]
```diff
@@ -7,6 +7,17 @@ ipfs_proxy_admin_port: ~
 
 ipfs_proxy_site_path: '/var/www/ipfs'
 
+# for rate limiting
+ipfs_proxy_rate_limit: true
```
Comment (jakubgs) [inline ansible/roles/ipfs-proxy/defaults/main.yml:11]: You left the unused defaults in.

**Review by jakubgs — APPROVED** [review]
LGTM, we can drop the CF stuff once we switch to our own SSL termination in separate PR.

Please make sure to squash the commits and clean up the commit message.

**Pull request merged**